src/System.IO.Abstractions.TestingHelpers/MockFileStream.cs

```diff
@@ -31,7 +31,8 @@ public class MockFileStream : MemoryStream
                     throw CommonExceptions.FileAlreadyExists(path);
                 }
 
-                var fileData = mockFileDataAccessor.GetFile(path);
+                fileData = mockFileDataAccessor.GetFile(path);
+                fileData.LastAccessTime = DateTime.Now;
                 fileData.CheckFileAccess(path, access);
```

Comment:
This change caught my eye, as it changes the LastAccessTime before the check if we even can access the file.
Have you checked what the real filesystem does in case of the code path covered in `fileData.CheckFileAccess(...)` ?

To make it obvious, it is not clear to me if `fileData.LastAccessTime = DateTime.Now;` should becomme before or after the line `fileData.CheckFileAccess(path, access);`

Reply:
Thank you for your thoughtful look! 
Obviously the check `fileData.CheckFileAccess(path, access);` should be done before a timestamp is updated. I also realised that the LastAccessTime timestamp should not be updated in the constructor, but only when the file is actually accessed. 

This has been fixed in d5f5d1e